### ISSUE TITLE
C/C++ visualization

### ISSUE THREAD

**Issue by khamyl** [hediet/vscode-debug-visualizer#119]
Hello,

I am so happy that something like this exists! Unfortunately, I cannot make it work for C++ (cppvsdbg). I took the example form [demo code](https://github.com/hediet/vscode-debug-visualizer/blob/master/demos/cpp/main.cpp) and implemented it like this:

```C++
std::string visualize = "{\
        \"kind\":{ \"plotly\": true },\
        \"data\":[\
            { \"y\": [1, 2, 4, 8, 16] },\
            { \"y\": [14, 3, 0, 15, 10] }\
        ]\
    }";
```
The result is as following:
![image](https://user-images.githubusercontent.com/4025815/125065509-32710400-e0b2-11eb-924e-a026a059fa1f.png)

How to create a JSON string in C++ that would be parsed as expected?

Related to #108

**Comment by hediet** [comment]
Interesting, that should work.

What does the debug console say when evaluating this expression there?

**Comment by khamyl** [comment]
```json
"{        \"kind\":{ \"plotly\": true },        \"data\":[            { \"y\": [1, 2, 4, 8, 16] },            { \"y\": [14, 3, 0, 15, 10] }        ]    }"
```

![image](https://user-images.githubusercontent.com/4025815/125077967-0a899c80-e0c2-11eb-8fb3-65ecd5ae209d.png)

**Comment by khamyl** [comment]
Any ideas?

**Comment by khamyl** [comment]
Is that a bug, or am I doing something incorrectly?

**Comment by hediet** [comment]
I'm sorry, I did not find time to look into it...
But I really want to.

I think this might be a bug.
If it is not a bug, the extension is not doing a great job at showing what is wrong, so in both cases, I need to look into it.

**Comment by khamyl** [comment]
May I help somehow?

**Comment by hediet** [comment]
Yeah sure, you should be able to debug the extension (see contributing).

I think this is a good code pointer:
https://github.com/hediet/vscode-debug-visualizer/blob/master/extension/src/VisualizationBackend/GenericVisualizationSupport.ts#L60

**Comment by nbwuzhe** [comment]
Hi @khamyl  May I ask if you have found some solution for this issue? Looks like I'm also facing the same problem.

**Comment by thijssteel** [comment]
I ran into the same issue and I think I found the problem and a solution.

The problem seems to be that evaluating 'visualise', just outputs the STL container and not the actual string. Enabling pretty printing solves that issue. ( `-exec -enable-pretty-printing` )

A second issue in the example above is that there are a lot of spaces in the string. GDB will shorten this to say `<repeated 12 times>` which the JSON parser cannot read. The easiest solution is to remove the spaces, although a different printing method may be necessary for things like "aaaaa".

This is related to https://github.com/microsoft/vscode-cpptools/issues/201

**Comment by hediet** [comment]
Do you have ideas how to improve this generally?

I noticed that debug adapters change their output from time to time.
I think it should be easier to debug that.

**Comment by thijssteel** [comment]
I don't really know.

Getting the full string is easier if you additionally use the commands:
`-exec set print elements 0`
`-exec set print repeats 0`

But that's not a true solution either. I think the best you can do in this library is add a warning in de cpp example about gdb's behaviour and these three commands that hopefully make it better.

**Comment by arjunarjun07** [comment]
I am facing the same with MSVC compiler in vscode. The debug visualizer displays the same output as the above. Any help would be appreciated.
